homeassistant/components/modbus/__init__.py

```diff
@@ -291,6 +312,25 @@
         ),
         vol.Optional(CONF_SWITCHES): vol.All(cv.ensure_list, [SWITCH_SCHEMA]),
         vol.Optional(CONF_FANS): vol.All(cv.ensure_list, [FAN_SCHEMA]),
+        vol.Optional(CONF_SCAN_GROUPS): vol.All(
+            cv.ensure_list,
+            [
+                vol.All(
+                    {
+                        vol.Required(CONF_NAME): cv.string,
+                        vol.Required(CONF_SCAN_INTERVAL_MILLIS): cv.positive_int,
+                    }
+                )
+            ],
+        ),
+        vol.Optional(CONF_ENOCEAN): vol.Maybe(
```

Comment:
I don't think we'll approve changes like this that adds specific details from another integration to an integration. If it can be abstracted to general purpose it could be considered if the code owners approves.

Reply:
I currently do not see this as a generic extension. The better solution is that you inherit the modbus classes you need in your integration with the extensions you need.

There are a couple of integrations that do it like that.

Reply:
Ah, extracting the Wago Modbus Enocean specific to a custom component might definitely make sense. Can you point me to an example component doing this already so I can have a look?